packages/tooltip/src/Tooltip.tsx

```diff
@@ -84,55 +84,78 @@ interface PopoverFunctionParameters {
 
 type ModifiedPopoverProps = Omit<PopoverProps, 'active' | 'spacing' | 'refEl'>;
 
-interface TooltipProps
-  extends Omit<HTMLElementProps<'div'>, keyof ModifiedPopoverProps>,
-    ModifiedPopoverProps {
-  /**
-   * A slot for the element used to trigger the `Tooltip`.
-   * default: hover
-   */
-  trigger: React.ReactElement | Function;
-
-  /**
-   * Determines if a `hover` or `click` event will trigger the opening of a `Tooltip`.
-   * default: 'hover'
-   */
-  triggerEvent?: TriggerEvent;
-
-  /**
-   * Controls component and determines the open state of the `Tooltip`
-   * default: `false`
-   */
-  open?: boolean;
-
-  /**
-   * Callback to change the open state of the `Tooltip`.
-   */
-  setOpen?: React.Dispatch<React.SetStateAction<boolean>>;
-
-  /**
-   * Whether the `Tooltip` will be `light` or `dark`.
-   * default: light
-   */
-  variant?: Variant;
-
-  /**
-   * id given to `Tooltip` content.
-   */
-  id?: string;
-
-  /**
-   * Callback to determine whether or not `Tooltip` should close when user tries to close it.
-   *
-   */
-  shouldClose?: () => boolean;
-
-  /**
-   * Enables Tooltip to trigger based on the event specified by `triggerEvent`.
-   * default: true
-   */
-  enabled?: boolean;
-}
+export type TooltipProps = Omit<
+  HTMLElementProps<'div'>,
+  keyof ModifiedPopoverProps
+> &
+  ModifiedPopoverProps & {
+    /**
+     * A slot for the element used to trigger the `Tooltip`.
+     * default: hover
+     */
+    trigger: React.ReactElement | Function;
+
+    /**
+     * Determines if a `hover` or `click` event will trigger the opening of a `Tooltip`.
+     * default: 'hover'
+     */
+    triggerEvent?: TriggerEvent;
+
+    /**
+     * Controls component and determines the open state of the `Tooltip`
+     * default: `false`
+     */
+    open?: boolean;
+
+    /**
+     * Callback to change the open state of the `Tooltip`.
+     */
+    setOpen?: React.Dispatch<React.SetStateAction<boolean>>;
+
+    /**
+     * Whether the `Tooltip` will be `light` or `dark`.
+     * default: light
+     */
+    variant?: Variant;
+
+    /**
+     * id given to `Tooltip` content.
+     */
+    id?: string;
+
+    /**
+     * Callback to determine whether or not `Tooltip` should close when user tries to close it.
+     *
+     */
+    shouldClose?: () => boolean;
+
+    /**
+     * Enables Tooltip to trigger based on the event specified by `triggerEvent`.
+     * default: true
+     */
+    enabled?: boolean;
+  } & (
+    | {
+        /**
+         * Specifies that the popover content will appear portaled to the end of the DOM,
+         * rather than in the DOM tree.
+         *
+         * default: `true`
+         */
+        usePortal?: true;
+
+        /**
+         * If using a portal, specifies a class name to apply to the root element of the portal.
+         *
+         * default: undefined
+         */
+        portalClassName?: string;
+      }
+    | {
+        usePortal: false;
+        portalClassName?: never;
```

Comment:
Not sure how many times we'll duplicate this, but if we see us doing it more we should extract to something reusable.